lib/core/src/sdk.rs

```diff
@@ -301,28 +305,40 @@ impl LiquidSdk {
     async fn check_chain_swap_expiration(&self, chain_swap: &ChainSwap) -> Result<()> {
         if chain_swap.user_lockup_tx_id.is_some() && chain_swap.refund_tx_id.is_none() {
             match chain_swap.direction {
+                Direction::Incoming => {
+                    let swap_script = chain_swap.get_lockup_swap_script()?.as_bitcoin_script()?;
+                    let current_height =
+                        self.bitcoin_chain_service.lock().await.tip()?.height as u32;
+                    let locktime_from_height =
+                        LockTime::from_height(current_height).map_err(|e| {
+                            PaymentError::Generic {
+                                err: format!(
+                                    "Error getting locktime from height {current_height:?}: {e}",
+                                ),
+                            }
+                        })?;
+
+                    info!("Checking Chain Swap {} expiration: locktime_from_height = {locktime_from_height:?},  swap_script.locktime = {:?}", chain_swap.id, swap_script.locktime);
+                    if swap_script.locktime.is_implied_by(locktime_from_height) {
+                        let id: &String = &chain_swap.id;
+                        info!("Chain Swap {} user lockup tx was broadcast. Setting the swap to refundable.", id);
+                        self.chain_swap_state_handler
+                            .update_swap_info(id, Refundable, None, None, None, None)
+                            .await?;
+                    }
+                }
```

Comment:
This step for refunding incoming chain swaps is a bit tricky since there are some edge cases:
1. We can refund several times (even though there was an historical refund_id) for example when a user has sent funds to the swap address after it has completed .
2. Also we enter here only for pending swaps and we need to check also historical swaps (although we can do that not at the same frequencey if we want to optimize performance).

In general I think we should mark refundable if the swap address has positive balance (can lwk wallet give us this information?) and the swap has expired.

Reply:
> In general I think we should mark refundable if the swap address has positive balance (can lwk wallet give us this information?) and the swap has expired.

In this incoming case the swap address is a BTC address. During the swap we rely on the boltz status to tell us if there is a tx going to this address.